src/test/ui/consts/issue-66693.rs

```diff
@@ -0,0 +1,25 @@
+// Tests that the compiler does not ICE when const-evaluating a `panic!()` invocation with a
+// non-`&str` argument.
+
+#![feature(const_panic)]
+
+const _: () = panic!(1);
+//~^ ERROR: argument to `panic!()` in a const context must have type `&str`
+
+static _FOO: () = panic!(true);
+//~^ ERROR: argument to `panic!()` in a const context must have type `&str`
+
+const fn _foo() {
+    panic!(&1); //~ ERROR: argument to `panic!()` in a const context must have type `&str`
+}
+
+// ensure that conforming panics are still handled correctly
+// const-evaluation halts before triggering these but not in the array-len case
```

Comment:
I do not understand the second part of this comment... it seems more confusing than helpful?